modules/fluid_properties/src/userobjects/Water97FluidProperties.C

```diff
@@ -0,0 +1,1411 @@
+      }
+      else if (temperature <= TSat(pressure))
+      {
+        if (pMPa > 21.93161551 && pMPa <= 22.064)
+          if (temperature > tempXY(pressure, "qu") && temperature <= tempXY(pressure, "uv"))
+            subregion = 20;
+          else
+            subregion = 24;
+        else // (pMPa > pSat(643.15) * 1.0e-6 && pMPa <= 21.93161551)
+          subregion = 20;
+      }
+      else if (temperature > TSat(pressure))
+      {
+        if (pMPa > 21.90096265 && pMPa <= 22.064)
+        {
+          if (temperature <= tempXY(pressure, "wx"))
+            subregion = 25;
+          else
+            subregion = 23;
+        }
+        else
+          subregion = 23;
+      }
+    }
+    else if (temperature > tempXY(pressure, "rx") && temperature <= tempXY(pressure, "jk"))
+      subregion = 17;
+    else
+      subregion = 10;
+  }
+  else if (pMPa > 20.5 && pMPa <= pSat(643.15) * 1.0e-6) // pSat(643.15) = 21.04 MPa
+  {
+    if (temperature <= tempXY(pressure, "cd"))
+      subregion = 2;
+    else if (temperature > tempXY(pressure, "cd") && temperature <= TSat(pressure))
+      subregion = 16;
+    else if (temperature > TSat(pressure) && temperature <= tempXY(pressure, "jk"))
+      subregion = 17;
+    else // (temperature > tempXY(pressure, "jk"))
+      subregion = 10;
+  }
+  else if (pMPa > P3cd && pMPa <= 20.5) // P3cd  = 19.00881189173929
+  {
+    if (temperature <= tempXY(pressure, "cd"))
+      subregion = 2;
+    else if (temperature > tempXY(pressure, "cd") && temperature <= TSat(pressure))
+      subregion = 18;
+    else
+      subregion = 19;
+  }
+  else if (pMPa > pSat(623.15) * 1.0e-6 && pMPa <= P3cd)
+  {
+    if (temperature < TSat(pressure))
+      subregion = 2;
+    else
+      subregion = 19;
+  }
+  else if (pMPa > 22.11 && pMPa <= 22.5)
+  {
+    if (temperature > tempXY(pressure, "qu") && temperature <= tempXY(pressure, "uv"))
+      subregion = 20;
+    else if (temperature > tempXY(pressure, "uv") && temperature <= tempXY(pressure, "ef"))
+      subregion = 21;
+    else if (temperature > tempXY(pressure, "ef") && temperature <= tempXY(pressure, "wx"))
+      subregion = 22;
+    else // (temperature > tempXY(pressure, "wx") && temperature <= tempXY(pressure, "rx"))
+      subregion = 23;
+  }
+  else if (pMPa > 22.064 && pMPa <= 22.11)
+  {
+    if (temperature > tempXY(pressure, "qu") && temperature <= tempXY(pressure, "uv"))
+      subregion = 20;
+    else if (temperature > tempXY(pressure, "uv") && temperature <= tempXY(pressure, "ef"))
+      subregion = 24;
+    else if (temperature > tempXY(pressure, "ef") && temperature <= tempXY(pressure, "wx"))
+      subregion = 25;
+    else // (temperature > tempXY(pressure, "wx") && temperature <= tempXY(pressure, "rx"))
+      subregion = 23;
+  }
+  else
+    mooseError("Water97FluidProperties::subregion3. Shouldn't have got here!");
+
+  return subregion;
+}
+
+Real
+Water97FluidProperties::tempXY(Real pressure, std::string xy) const
```

Comment:
Yeah, my really good reason was that I was lazy and just cut and pasted what I had done a few years ago. I'll swap this over to an enum.